### ISSUE TITLE
Add FixQualityIntValue property in NmeaSentence to preserve the parsed fix quality int value from the sentence

### ISSUE THREAD

**Issue by gojanpaolo** [DotSpatial/DotSpatial#888]
Could we add a property in the NmeaSentence base class to preserve the real int value of the fix quality?
`(int)FixQuality ` will not work properly since the int value and the description varies within different sentences.

We could also upgrade the FixQuality enumeration into a class/struct that will store its value and description.

**Comment by mogikanin** [comment]
@gojanpaolo can you show code sample with issue? I'm a bit confused, what exactly is wrong :)
In your code you can't rely on int representation of FixQuality values, since they not necessary correlated to integers in GPS sentences. As you might seen inside NmeaSentence.cs there are a switches to parse int positions into FixQuality values
https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Positioning/NmeaSentence.cs#L217
https://github.com/DotSpatial/DotSpatial/blob/master/Source/DotSpatial.Positioning/NmeaSentence.cs#L254

**Comment by gojanpaolo** [comment]
Yes, there is an int to FixQuality parse method in NmeaSentence.cs but not the other way around. So let's say I made a code like this.

```
var nmeaSentence = nmeaReader.ReadTypedSentence();
var fixQuality = (nmeaSentence as IFixQualitySentence)?.FixQuality ?? FixQuality.Unknown;
//I have the FixQuality enum value but the corresponding integer value of it is not preserved.
//also, please note that there are other sentences (e.g. [Trimble Sentences](https://www.trimble.com/OEM_ReceiverHelp/V4.44/en/NMEA-0183messages_PTNL_GGK.html)) that do not have the same int to FixQuality conversions.
```

**Comment by mogikanin** [comment]
Yes, they not same. So you need to write own conversion methods.
